roads.mss

```diff
@@ -3106,3 +3106,111 @@ tertiary is rendered from z10 and is not included in osm_planet_roads. */
     }
   }
 }
+#railways-text-name {
+  /* Mostly started from z17. */
+  [railway = 'rail'],
+  [railway = 'subway'],
+  [railway = 'narrow_gauge'],
+  [railway = 'light_rail'],
+  [railway = 'preserved'],
+  [railway = 'funicular'],
+  [railway = 'monorail'],
+  [railway = 'turntable'],
+  [railway = 'tram'] {
+    [zoom >= 17] {
```

Comment:
You need to add something like [highway = null] to avoid road names repeated as railway labels.